### ISSUE TITLE
Latest openshift-ansible fails the origin-gce build

### ISSUE THREAD

**Issue by smarterclayton** [openshift/openshift-ansible#3843]
Sometime after march 22nd, the origin-gce job started failing https://ci.openshift.redhat.com/jenkins/job/zz_origin_gce_image/205/ due to:

```
TASK [openshift_version : fail] ************************************************
task path: /usr/share/ansible/openshift-ansible/roles/openshift_version/tasks/main.yml:68
Friday 31 March 2017  11:32:58 +0000 (0:00:00.049)       0:05:13.219 ********** 
fatal: [ci-primg205-ig-m-fk2c]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "msg": "No OpenShift version available, please ensure your systems are fully registered and have access to appropriate yum repositories."
        }, 
        "module_name": "fail"
    }, 
    "msg": "No OpenShift version available, please ensure your systems are fully registered and have access to appropriate yum repositories."
}
```

The last image we pushed has no such problems, so it's likely a change in openshift-ansible.  The RPMs we are using are:

https://storage.googleapis.com/origin-ci-test/branch-logs/origin/master/builds/147/artifacts/rpms/origin-3.6.0-0.alpha.0.58.40bcfef.x86_64.rpm

I kicked another run to see if it resolved itself (https://ci.openshift.redhat.com/jenkins/job/zz_origin_gce_image/206) but won't be online to investigate.

@stevekuznetsov @sdodson

**Comment by sdodson** [comment]
For some reason the `deployment_type=origin` but `openshift_deployment_type=openshift-enterprise` so it's looking for enterprise package names, finding none and setting the version to `0.0`.

@sosiouxme may have a clue

**Comment by sosiouxme** [comment]
Yeah, I see what's happening. This was just after my `openshift_sanitize_inventory` role went in, and it's working. With both `deployment_type=origin` and `openshift_deployment_type=openshift-enterprise` set, `openshift_deployment_type` now wins, where before it would not have overridden `deployment_type`. It didn't occur to me someone might set both and to different values.

So `openshift-enterprise` is the effective deployment type and things fail as there are no enterprise pkgs. Simple solution is to remove `deployment_type=origin` and set `openshift_deployment_type=origin` in the inventory. Having both set to different values is a bug anyway.

**Comment by sdodson** [comment]
I'd hope they're not actually set to different values in the inventory, @smarterclayton can you point @sosiouxme at the variables set for this install?

**Comment by stevekuznetsov** [comment]
`deployment_type` is set [here](https://github.com/openshift/release/blob/master/cluster/test-deploy/data/vars.yaml#L3)

@sosiouxme do you expect us to migrate from `deployment_type` to `openshift_deployment_type`? (in general, not for this specific case)

**Comment by smarterclayton** [comment]
Isn't that breaking for old scripts?

**Comment by sdodson** [comment]
It passed both aos-ci-test and the merge job which both use `deployment_type` rather than `openshift_deployment_type` so one would hope, but something is obviously amiss.

**Comment by mbarreca** [comment]
I'm having the same issue when I'm trying to install this on a few GCE instances. 

Weirdly masters 2,3, loadbalancer 1 and the node works fine but master 1 can't handle it. I wiped it and remade the master to no avail

```
TASK [openshift_version : fail] ************************************************
fatal: [os1.*****.com]: FAILED! => {"changed": false, "failed": true, "msg": "No OpenShift version available, please ensure your systems are fully registered and have access to appropriate yum repositories."}
        to retry, use: --limit @/root/openshift-ansible/playbooks/byo/config.retry
```


```
# Create an OSEv3 group that contains the master, nodes, etcd, and lb groups.
# The lb group lets Ansible configure HAProxy as the load balancing solution.
# Comment lb out if your load balancer is pre-configured.
[OSEv3:children]
masters
nodes
etcd
lb

# Set variables common for all OSEv3 hosts
[OSEv3:vars]
ansible_ssh_user=root
openshift_deployment_type=openshift-enterprise

# Uncomment the following to enable htpasswd authentication; defaults to
# DenyAllPasswordIdentityProvider.
openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true', 'challenge': 'true', 'kind': 'HTPasswdPasswordIdentityProvider', 'filename': '/etc/origin/master/htpasswd'}]

# Native high availbility cluster method with optional load balancer.
# If no lb group is defined installer assumes that a load balancer has
# been preconfigured. For installation the value of
# openshift_master_cluster_hostname must resolve to the load balancer
# or to one or all of the masters defined in the inventory if no load
# balancer is present.
openshift_master_cluster_method=native
openshift_master_cluster_hostname=osc..*****.com
openshift_master_cluster_public_hostname=osc..*****.com

# override the default controller lease ttl
#osm_controller_lease_ttl=30

# host group for masters
[masters]
os1.*****.com
os2.*****.com
os3.*****.com

# host group for etcd
[etcd]
os1.*****.com
os2.*****.com
os3.*****.com

# Specify load balancer host
[lb]
lb..*****.com

# host group for nodes, includes region info
[nodes]
os[1:3].*****.com openshift_node_labels="{'region': 'infra', 'zone': 'default'}"
node1.*****.com openshift_node_labels="{'region': 'primary', 'zone': 'east'}"
```

**Comment by sosiouxme** [comment]
>
>
> @sosiouxme <https://github.com/sosiouxme> do you expect us to migrate
> from deployment_type to openshift_deployment_type? (in general, not for
> this specific case)
>

In general, yes, but both should work until we decide we don't want that.

I say both are getting set because that's what I see in the debug output
from the run... they are set to something different until
openshift_sanitize_inventory sets them the same. I don't see in the
inventory how they're getting set differently to begin with, though.

**Comment by sosiouxme** [comment]
@stevekuznetsov technically https://github.com/openshift/release/blob/master/cluster/test-deploy/data/vars.yaml is just being added to whatever vars are set in the inventory or elsewhere. *Something* is setting `openshift_deployment_type` to `openshift-enterprise`.

From [this run](https://ci.openshift.redhat.com/jenkins/job/zz_origin_gce_image/206/consoleFull), looks like it's this:

```
TASK [cluster-variables : include_vars] ****************************************
task path: /usr/share/ansible/openshift-ansible-gce/playbooks/roles/cluster-variables/tasks/main.yaml:1
Monday 03 April 2017  23:09:30 +0000 (0:00:01.172)       0:03:14.419 ********** 
ok: [ci-primg206-ig-m-hkdj] => {
    "ansible_facts": {
        "_cluster_variables": {
            ...
            "deployment_type": "origin", 
            ...
            "openshift_deployment_type": "openshift-enterprise", 
```
Which comes from https://github.com/openshift/origin-gce/blob/master/playbooks/roles/cluster-variables/vars/main.yaml#L20

I'm not sure how the precedence works out, but setting `openshift_deployment_type` instead of `deployment_type` may still fix it.

**Comment by stevekuznetsov** [comment]
Yeah, that file at [`origin-gce/playbooks/roles/cluster-variables/vars/main.yaml`](https://github.com/openshift/origin-gce/blob/master/playbooks/roles/cluster-variables/vars/main.yaml#L20) looks wrong. Not sure why that exists. If you do make a change to migrate `deployment_type` please let us know and we'll update as necessary. For now I'll change that var.

**Comment by smarterclayton** [comment]
I think we should probably return an error if they are different (from the fact setup), if it's an invalid setup.

**Comment by sosiouxme** [comment]
That could be done easily enough. Having them set differently is probably a
mistake and is now interpreted differently than before.

**Comment by smarterclayton** [comment]
Fixed by #3902
